FaissMask/Internal/IndexSafeHandle.cs

```diff
@@ -79,17 +81,19 @@ protected override bool ReleaseHandle()
             return true;
         }
 
-        public float[] ReconstructVector(long key, int dimensions)
+        public float[] ReconstructVector(long key)
         {
+            var dimensions = NativeMethods.faiss_Index_d(this);
```

Comment:
Can this be `var dimensions = Dimensions` so we re-use the property definition instead?
```suggestion
            var dimensions = Dimensions;
```

Reply:
Updated